### ISSUE TITLE
[Bug]: Refresh is to high

### ISSUE THREAD

**Issue by fapgomes** [nurikk/z2m-frontend#2288]
### What happened?

Hi,

 When using the zigbee2mqtt frontend, the refresh rate is so high that it makes it very difficult to use the frontend. In the attached video, I try to click several times on the water_leak_detector.

https://github.com/user-attachments/assets/56b4ad6d-a21e-4360-a294-8cb40d1f3da5

### What browsers are you seeing the problem on?

Firefox, Chrome

### Did you download state.json.zip?

I will attach state.json.zip

**Comment by bluemoehre** [comment]
Is this still up-to-date? Normally React should prevent re-rendering the DOM if there are no changes. But depending on the architecture of that view this might be a structural bug if continuous changes in one item cause the whole table to re-render.

EDIT:
Looks like the first item gets updates every second. Is that one causing it?

**Comment by starox** [comment]
I have a bad user experience because of the refresh rate: I can't select text on device list to do copy and paste.
I think that I have too many devices and so refresh is too frequent within one second.
I must go on a less busy page to have enough time to do mouse selection.

I wonder if it is possible to have a minimum amount of time between rendering and ... ideally on some part of the site ?

**Comment by bluemoehre** [comment]
Okay, so I assume the bug is still there.

As said, with a optimal React structure only changed Cells / Rows are re-rendered, so you won't loose focus / text selection etc. in other elements. 

The devs need to check if indexes of components in loops are correct and the component structure is build in a way React is able to re-render with a minimum impact on elements.

EDIT:
to workaround you may disable the "last seen" + "availability" feature. That is causing lots of updates.

**Comment by starox** [comment]
I have availability enable (I wan't this ;-) ) and 120 devices.

Oddly, when I look mosquitto on my desktop with [mqtt-explorer](https://mqtt-explorer.com/), I have ~10 events per seconds.
When looking on my smartphone with [MQTTapp](https://play.google.com/store/apps/details?id=com.vielsack.mqtt&hl=en-US&pli=1), I have ~100 events per seconds.
I will refine avaibility device per device later to ease the system.

Is there a way in zigbee2mqtt to monitor zigbee traffic, ideally making itself as a discovery device for home-assistant ?

Another idea, make a "freeze/unfreeze" button for the interface ?

**Comment by fapgomes** [comment]
> Okay, so I assume the bug is still there.

Yes is here.

> As said, with a optimal React structure only changed Cells / Rows are re-rendered, so you won't lose focus/text selection etc. in other elements.

Right now, I lose focus/text selection.

> EDIT: to workaround you may disable the "last seen" + "availability" feature. That is causing lots of updates.

I use that to see if they are available :) 

Best Regards,
./fapg

**Comment by bluemoehre** [comment]
FYI:
I only have ~90 devices at the moment and I can't reproduce that issue even when enabling 'Availability' + 'Last seen' by myself with that intensity like in the video. 
While no lights are changed, I got about 2-20 events per minute - this seems more 'normal' to me. 

So, besides the UI bug, you guys maybe also want to check for unnecessary polling / binds active or really bad devices which are very talkative? That amount of communication will also drain your battery-powered devices.

**Comment by starox** [comment]
This seems similar to my setup.
My use case is that:
- go to device (first item on top list)
- try to select completely an ieee address (fourth column)
Usually I can't right click and select copy, the selection removes itself very fast.
The line where you click don't necessary changes.

So I wonder if it is related to the browser removing selection as soon as another line changes ?
I am using firefox 128.06esr on amd64

**Comment by bluemoehre** [comment]
@starox no, I can confirm **it's a bug in the UI**. If you check the view using React DevTools you will see the whole table including the search box is re-rendered frequently (each time the data in the table changes). 

**Workaround**: \
Try to get less data changes by e.g. disabling features

**Quickfix**: \
Devs needs to implement a [debouncer](https://www.npmjs.com/package/use-debounce)

**Fix**: \
The view needs to be fully checked to find out where in the component structure item keys are missing and/or where references get lost, making React unable to replace only updated cells/content. 

---

**Reproduction:**

- enable 'Last seen' + 'Availability' cols
- start any device firmware update

**Result:**
It is nearly impossible to click any link in the table because of hundreds of re-renders per second. \
Also it's causing high CPU load in the Tab.

**Expectation:**
Only changed contents should get re-rendered / repainted by React.

---

@nurikk are you able to provide at least a hotfix within the next weeks? Unfortunately I cannot setup the project on my Windows machine due to multiple undocumented issues & requirements (maybe you can help with that as well), but at least I can help with the debugging or debouncing.

**Comment by bluemoehre** [comment]
I tried fixing this in the code, but I ran into endless setup issues, struggled to get it running, and honestly, understanding everything was a nightmare, due to the complexity / API with a foreign architecture etc ... so, well… uh… plz don't blame me, but ... I just started rewriting the whole app. 😅

<img width="800" alt="Image" src="https://github.com/user-attachments/assets/f49d50a5-894d-4ac4-acb3-1086440a4465" />

If you're interested, I can deploy the UI somewhere so you can connect directly to your backend in your browser. I'll publish the source later, once most views are functional.

**Comment by Koenkk** [comment]
@bluemoehre looks really nice!

**Comment by bluemoehre** [comment]
@Koenkk I don’t want to derail the current topic, but is there a way I could get in touch with you to discuss the API and typings? I've spent weeks gathering information across various related projects, but still ended up having to reverse-engineer some endpoints. I also have a few suggestions that I believe could help simplify things.
As a thank-you for your time and effort, I'd be happy to offer you an early alpha sneak peek =D

**Comment by Koenkk** [comment]
@bluemoehre yes, feel free to contact me at the z2m discord.

**Comment by fapgomes** [comment]
I still have the problem, but it is not a major bug.
